src/modules/launcher/Plugins/Microsoft.Plugin.WindowWalker/Components/WindowWalkerSettings.cs

```diff
@@ -0,0 +1,164 @@
+                    Value = false,
+                },
+                new PluginAdditionalOption
+                {
+                    Key = nameof(SubtitleShowDesktopName),
+                    DisplayLabel = Resources.wox_plugin_windowwalker_SettingSubtitleDesktopName,
+                    Value = true,
+                },
+                new PluginAdditionalOption
+                {
+                    Key = nameof(ConfirmKillProcess),
+                    DisplayLabel = Resources.wox_plugin_windowwalker_SettingConfirmKillProcess,
+                    Value = true,
+                },
+                new PluginAdditionalOption
+                {
+                    Key = nameof(HideKillProcessOnElevatedProcesses),
+                    DisplayLabel = Resources.wox_plugin_windowwalker_SettingHideKillProcess,
+                    Value = false,
+                },
+                new PluginAdditionalOption
+                {
+                    Key = nameof(HideExplorerSettingInfo),
+                    DisplayLabel = Resources.wox_plugin_windowwalker_SettingExplorerSettingInfo,
+                    Value = false,
+                },
+            };
+
+            return optionList;
+        }
+
+        /// <summary>
+        /// Update this settings.
+        /// </summary>
+        /// <param name="settings">The settings for all power launcher plugins.</param>
+        internal void UpdateSettings(PowerLauncherPluginSettings settings)
+        {
+            if (settings is null || settings.AdditionalOptions is null)
+            {
+                return;
+            }
+
+            ResultsFromVisibleDesktopOnly = GetSettingOrDefault(settings, nameof(ResultsFromVisibleDesktopOnly));
+            SubtitleShowPid = GetSettingOrDefault(settings, nameof(SubtitleShowPid));
+            SubtitleShowDesktopName = GetSettingOrDefault(settings, nameof(SubtitleShowDesktopName));
+            ConfirmKillProcess = GetSettingOrDefault(settings, nameof(ConfirmKillProcess));
+            HideKillProcessOnElevatedProcesses = GetSettingOrDefault(settings, nameof(HideKillProcessOnElevatedProcesses));
+            HideExplorerSettingInfo = GetSettingOrDefault(settings, nameof(HideExplorerSettingInfo));
+        }
+
+        /// <summary>
+        /// Return one <see cref="bool"/> setting of the given settings list with the given name.
+        /// </summary>
+        /// <param name="settings">The object that contain all settings.</param>
+        /// <param name="name">The name of the setting.</param>
+        /// <returns>A settings value.</returns>
+        private static bool GetSettingOrDefault(PowerLauncherPluginSettings settings, string name)
+        {
+            // As a fallback if a setting isn't available, we use the value defined in the method GetAdditionalOptions()
+            var option = settings.AdditionalOptions.FirstOrDefault(x => x.Key == name);
```

Comment:
@TobiasSekan
Is it correct that `option` can only be null if the setting isn't defined as `PluginAdditionalOption`? Then this fallback feature makes no sense and we can fallback to `default(bool)` instead or remove fallback.

If I delete setting in json and start pt run I will get the default value from `PluginAdditionalOption` element.